protocol/opcm-test-integration.md

```diff
@@ -0,0 +1,130 @@
+    `/deployments/{chainId}-deploy.json`
+- EVM state:
+  - The test values configured by each test suite must be preserved (otherwise the diff to update
+    each test will explode).
+
+Nice to haves:
+- logs emitted during deployment:
+  - This is not essential, but comparing before and after logs is a quick and dirty way to ensure
+    that the config is roughly the same.
+- Deploying contracts to the same address as before.
+  - This might be achievable based on the use of
+    [`create2`](https://github.com/ethereum-optimism/optimism/blob/develop/packages/contracts-bedrock/scripts/deploy/Deploy.s.sol#L1703)
+    in the existing scripts. If so it would enable us to use statedumps to diff the before and after
+    deployments.
+
+## Proposed Solution
+
+A plausible PR stack for achieving this might be:
+
+1. Remove the `viaSafe` functionality: `Deploy.s.sol` currently starts our by deploying a new
+  `SystemOwnerSafe`, and then uses `callViaSafe()` and `upgradeAndCallViaSafe()` for all calls to
+    the `ProxyAdmin` While maintaining the `viaSafe()` helper methods (they are also needed in
+  `DeployOwnership` for now), we should remove all usages in `Deploy.s.sol`, perform all actions
+    through the deployer key, then finally transfer the `ProxyAdmin` to the `finalSystemOwner`
+    (which the current scripts don't even do at the moment).
+2. Deploy a separate `ProxyAdmin` contract for the Superchain. A key diff between the OPCM approach
+    and `Deploy.s.sol` is that the latter shares the same `ProxyAdmin` for the Superchain and the OP
+    Chain that it deploys. That needs to be fixed to prepare for the OPCM.
+3. Deploy the Superchain contracts using the OPCM's `DeploySuperchain.s.sol`. With the previous PRs,
+    this will hopefully be straight-forward.
+4. Deploy the OP Chain Contracts using the `OPContractsManager` contract. This will be much simpler
+    if we can delete the `OptimismPortal`, `L2OutputOracle` and `useFaultProofs` flag. Otherwise we
+    will accomplish this using two entirely distinct flows depending on the value of
+    `useFaultProofs`:
+      1. `true` -> deploy using OPCM
+      2. `false` -> deploy using the existing `deploy{ContractName}()` functions. It maybe possible
+         to reuse some of the contracts deployed by the OPCM which do not vary pre and post fault
+         proofs (ie. the `L1StandardBridge` and `L1CrossDomainMessenger`). This would allow us to
+         delete the deploy functions for those contracts, and will be investigated during
+         development.
+
+## Appendix
+
+The following content was created while investigating the current test setup flow, and
+is provided here for reference.
+
+##### Summary of the current Test Setup
+
+The Test contracts (`UltimateTestHarness` below) inherit from `CommonTest`, and implement a
+`setUp()` method which calls to `CommonTest.setUp()`. This results in an approximate call flow of:
```

Comment:
What does "approximate" mean here?

Reply:
Just that I've distilled it down to the most important details and added my own annotations which would not appear in a normal trace.